src/cmd/manage_nsfs.js

```diff
@@ -395,111 +389,75 @@ async function fetch_existing_account_data(action, target, decrypt_secret_key) {
     return data;
 }
 
+/**
+ * add_account creates a new account config file, 
+ * @param {Object} data
+ * @returns { Promise<{ code: typeof ManageCLIResponse.AccountCreated, detail: Object, event_arg: Object  }>} 
+ */
 async function add_account(data) {
-    await manage_nsfs_validations.validate_account_args(config_fs, data, ACTIONS.ADD, undefined);
-
-    const access_key = has_access_keys(data.access_keys) ? data.access_keys[0].access_key : undefined;
-    const name_exists = await config_fs.is_account_exists_by_name(data.name);
-    const access_key_exists = access_key && await config_fs.is_account_exists_by_access_key(access_key);
-
-    const event_arg = data.name ? data.name : access_key;
-    if (name_exists || access_key_exists) {
-        const err_code = name_exists ? ManageCLIError.AccountNameAlreadyExists : ManageCLIError.AccountAccessKeyAlreadyExists;
-        throw_cli_error(err_code, event_arg, {account: event_arg});
-    }
     data._id = mongo_utils.mongoObjectId();
-    const encrypted_account = await nc_mkm.encrypt_access_keys(data);
-    data.master_key_id = encrypted_account.master_key_id;
-    const encrypted_data = JSON.stringify(encrypted_account);
-    data = _.omitBy(data, _.isUndefined);
-    // We take an object that was stringify
-    // (it unwraps ths sensitive strings, creation_date to string and removes undefined parameters)
-    // for validating against the schema we need an object, hence we parse it back to object
-    const account = encrypted_data ? JSON.parse(encrypted_data) : data;
-    nsfs_schema_utils.validate_account_schema(account);
-    await config_fs.create_account_config_file(account);
-    write_stdout_response(ManageCLIResponse.AccountCreated, data, { account: event_arg });
+    await config_fs.create_account_config_file(data);
+    return { code: ManageCLIResponse.AccountCreated, detail: data, event_arg: { account: data.name } };
 }
 
-
-async function update_account(data, is_flag_iam_operate_on_root_account) {
-    await manage_nsfs_validations.validate_account_args(config_fs, data, ACTIONS.UPDATE, is_flag_iam_operate_on_root_account);
-
+/**
+ * update_account does the following updates an existing config file, 
+ * if a new_name provided it will pass old_name to the update function for unlinking the index
```

Comment:
```suggestion
 * if a new_name provided it will pass old_name to the update function for unlinking the index (symlink inside accounts_by_name/ dir)
```